docs/topics/items.rst

```diff
@@ -8,29 +8,116 @@ Items
+sources, typically, web pages. :ref:`Spiders <topics-spiders>` may return the
+extracted data as `items`, Python objects that define key-value pairs.
+
+Scrapy supports :ref:`multiple types of items <item-types>`. When you create an
+item, you may use whichever type of item you want. When you write code that
+receives an item, your code should :ref:`work for any item type
+<supporting-item-types>`.
+
+.. _item-types:
+
+Item Types
+==========
+
+Scrapy supports the following types of items: :ref:`dictionaries <dict-items>`,
+:ref:`dataclass objects <dataclass-items>`, :ref:`Item objects <item-objects>`,
+and :ref:`custom items <custom-items>`.
+
+.. _dict-items:
+
+Dictionaries
+------------
+
+As an item type, :class:`dict` is convenient and familiar.
+
+
+.. _dataclass-items:
+
+Dataclass objects
+-----------------
+
+.. versionadded:: 2.1
+
+:func:`~dataclasses.dataclass` allows defining item classes with field names,
+so that :ref:`item exporters <topics-exporters>` can export all fields by
+default even if the first scraped object does not have values for all of them.
+
+Dataclasses also allow defining the type and default value of each defined
+field.
+
+.. note:: Field names and types are not enforced at run time.
+
+They work natively in Python 3.7 or later, or using the `dataclasses
+backport`_ in Python 3.6.
+
+.. _dataclasses backport: https://pypi.org/project/dataclasses/
+
+
+.. _item-objects:
+
+Item objects
+------------
+
+:class:`Item` provides a :class:`dict`-like API plus additional features that
+make it the most feature-complete item type:
+
+.. class:: Item([arg])
+
+    :class:`Item` objects replicate the standard :class:`dict` API, including
+    its ``__init__`` method.
+
+    :class:`Item` allows defining field names, so that:
+
+    -   :class:`KeyError` is raised when using undefined field names (i.e.
+        prevents typos going unnoticed)
+
+    -   :ref:`Item exporters <topics-exporters>` can export all fields by
+        default even if the first scraped object does not have values for all
+        of them
+
+    :class:`Item` also allows defining field metadata, which can be used to
+    :ref:`customize serialization <topics-exporters-field-serialization>`.
+
+    :mod:`trackref` tracks :class:`Item` objects to help find memory leaks
+    (see :ref:`topics-leaks-trackrefs`).
+
+    :class:`Item` objects also provide the following additional API members:
+
+    .. automethod:: copy
+
+    .. automethod:: deepcopy
+
+    .. attribute:: fields
+
+        A dictionary containing *all declared fields* for this Item, not only
+        those populated. The keys are the field names and the values are the
+        :class:`Field` objects used in the :ref:`Item declaration
+        <topics-items-declaring>`.
+
+
+.. _custom-items:
+
+Custom items
+------------
+
+Subclass :class:`BaseItem` to define additional item types not based on any of
+the above:
+
+.. autoclass:: scrapy.item.BaseItem
```

Comment:
@Gallaecio @kmike I wonder if we should discourage inheriting from `BaseItem`, since it only inherits from `object_ref` and it doesn't provide any additional functionality?

Reply:
@elacuesta as we discussed today: this makes sense, but it'd be best if we tackle  it separately, and keep BaseItem checks for itemadapter.

Reply:
I deprecated `BaseItem` in https://github.com/scrapy/scrapy/pull/3881/commits/23c1739965119df3d91f5e288520c8c21e86312e, do you think it would be too much out of the scope of this PR? Opening a new PR would be fine too, I just think it'd be good to have it deprecated in 2.2 instead of waiting a new full version bump.

Reply:
Moved to a separate PR: #4534